HostTech DNS records managing module

felixfontein opened the pull request:
##### SUMMARY
This adds a new module: `hosttech_dns_record`, which allows to manage DNS records for zones hosted by the Swiss provider [hosttech](https://www.hosttech.ch/).

##### ISSUE TYPE
New Module Pull Request

##### COMPONENT NAME
hosttech_dns_record

##### ANSIBLE VERSION
devel (2.5.0)

##### ADDITIONAL INFORMATION
The module's interface is similar to `route53`'s interface, and allows to add, remove and update DNS records.

felixfontein committed:
Added HostTech DNS records managing module.

resmo commented on lib/ansible/modules/cloud/hosttech/hosttech_dns_record.py:
```diff
@@ -0,0 +1,1192 @@
+            records.append(record)
+
+    # Process get
+    if module.params.get('state') == 'get':
```
By policy, mixing a get or list function into a "normal" module is not accepted. A list or get function should be turned into a proper facts module.

felixfontein replied:
Ok, makes sense. I guess the `route53` module has been like that longer than this policy. I'll split the module up into two, then. Is it OK if the second module is in this PR as well, or should it be a separate PR?

resmo replied:
Yes, this policy is younger than route53.
Each module has to be in a separate pr, however this is sometimes difficult if you also have shared code for module_utils.

felixfontein replied:
There will be some shared code, since most of the API functionality will be needed by both modules. I'll split this up first, and then we can decide later whether to move the fact gathering module to another (later) PR or not.

felixfontein replied:
Ok, I split it up. Now ansibot complains about two modules in the PR, as (almost) expected ;)
There's another thing I noticed: could it be that the module should better be in the net_tools category (https://docs.ansible.com/ansible/latest/list_of_net_tools_modules.html)? If yes: should I move it around in this PR, or close this PR and open a new one?

felixfontein commented:
I put the facts retrieval module into another branch (which is based on this one): https://github.com/felixfontein/ansible/tree/hosttech-dns-record-facts

The `hosttech-dns-record-facts` branch slightly updates this module (adding another example) and adds a second, related module (`hosttech-dns-record-facts`) which shares most of the API-specific code (in `module_utils`).

One thing which isn't clear to me is whether `modules/cloud/hosttech` is a good place for this module, or whether `modules/net_tools/hosttech` would be better, since these modules don't do anything cloud-specific. (I first chose `cloud` because the DNS modules I knew were there; only later I noticed some in `net_tools` as well.)

felixfontein commented:
@resmo: can you comment on whether this should better be in `net_tools`?

resmo commented:
@felixfontein there is no hard policy, but if "hostech" has more than one API and you plan to add more modules, then "cloud" would be fine. If there is only an API for DNS, I would place it into "net_tools".

felixfontein commented:
@resmo: I'm only aware of the DNS API. Should I open a new PR for `net_tools`, or change this one?

resmo commented:
@felixfontein no need for a new PR, but whatever is comfortable for your ;)

felixfontein commented:
Ok. I rebased the module against current devel, updated `version_added` from 2.5 to 2.6, and moved it  from `modules/cloud/hosttech` to `modules/net_tools/hosttech`. I also adjusted the license of the `module_utils` part to BSD 2-clause (copied header from other modules in there), and updated the header of the document fragments to be more similar to most of the other files.
